src/Microsoft.Data.Analysis/DataFrame.IO.cs

```diff
@@ -198,55 +198,58 @@ private static DataFrameColumn CreateColumn(Type kind, string[] columnNames, int
                                 Encoding encoding = null)
         {
             if (!csvStream.CanSeek)
+            {
                 throw new ArgumentException(Strings.NonSeekableStream, nameof(csvStream));
+            }
 
             var linesForGuessType = new List<string[]>();
             long rowline = 0;
             int numberOfColumns = dataTypes?.Length ?? 0;
 
             if (header == true && numberOfRowsToRead != -1)
+            {
                 numberOfRowsToRead++;
+            }
 
             List<DataFrameColumn> columns;
             long streamStart = csvStream.Position;
             // First pass: schema and number of rows.
             using (var streamReader = new StreamReader(csvStream, encoding ?? Encoding.UTF8, detectEncodingFromByteOrderMarks: true, DefaultStreamReaderBufferSize, leaveOpen: true))
             {
                 string line = null;
-                if (dataTypes == null)
```

Comment:
Do you remember why we were doing this in the first place? Or was this just a mistake?

Reply:
In the case of `dataTypes` being passed in, should we stop processing after the header row?

---
In reply to: [416981821](https://github.com/dotnet/corefxlab/pull/2907#discussion_r416981821) [](ancestors = 416981821)

Reply:
Yea I was wondering about it too. I think I was using dataTypes inside as part of "guessing" the types at one point. Either that or it was a mistake made from trying to optimize the csv read.

Reply:
Good idea. I fixed that, and found some other corner cases and added unit tests for them.